dali/operators/geometry/affine_transforms/shear_transform.cc

```diff
@@ -0,0 +1,176 @@
+// Copyright (c) 2020, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "dali/operators/geometry/affine_transforms/transform_base_op.h"
+#include "dali/pipeline/data/views.h"
+#include "dali/core/geom/transform.h"
+#include "dali/core/math_util.h"
+
+namespace dali {
+
+DALI_SCHEMA(ShearTransform)
+  .DocStr(R"code(Produces a shear affine transform matrix.
+
+If another transform matrix is passed as an input, the operator applies the shear mapping to the matrix provided.
+
+.. note::
+    The output of this operator can be fed directly to the ``MT`` argument of ``CoordTransform`` operator.
+)code")
+  .AddOptionalArg<std::vector<float>>(
+    "shear",
+    R"code(The shear factors.
+
+For 2D, ``shear`` contains two elements; shear_xy, shear_yx.
+
+For 3D, ``shear`` contains six elements: shear_xy, shear_xz, shear_yx, shear_yz, shear_zx, shear_zy.
+
+A shear factor value can be interpreted as the offset to be applied in the first axis when moving in the
+direction of the second axis.
+
+.. note::
+    This argument is mutually exclusive with ``angles``. 
+    If provided, the number of dimensions of the transform is inferred from this argument.
+)code",
+    nullptr, true)
+  .AddOptionalArg<std::vector<float>>(
+    "angles",
+    R"code(The shear angles, in degrees.
+
+This argument is mutually exclusive with ``shear``.
+
+For 2D, ``angles`` contains two elements: angle_xy, angle_yx.
+
+For 3D, ``angles`` contains six elements: angle_xy, angle_xz, angle_yx, angle_yz, angle_zx, angle_zy.
+
+A shear angle is translated to a shear factor as follows::
```

Comment:
```suggestion
A shear angle is translated to a shear factor as follows::

```
I guess you need that newline to get a well-formated resultin sphinx.